### PR TITLE
host.json should not be patched when function names are empty #488

### PR DESCRIPTION
**Pull request by maartenba**

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/run/HostJsonPatcher.kt [hunk 1]
```diff
@@ -42,26 +42,34 @@ object HostJsonPatcher {
...
+    fun tryPatchHostJsonFile(workingDirectory: String, functionNames: String) =
+            tryPatchHostJsonFile(
+                    hostJsonFile = determineHostJsonFile(workingDirectory),
+                    functionNames = functionNames.split(',', ';', '|', ' '))
 
-        val hostJsonFile = determineHostJsonFile(workingDirectory)
+    private fun tryPatchHostJsonFile(hostJsonFile: File?, functionNames: List<String>) {
 
-        if (hostJsonFile == null) {
+        if (hostJsonFile == null || !hostJsonFile.exists()) {
```
Comment (sdubov) [inline PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/run/HostJsonPatcher.kt:52]: No changes needed, just checking - this should be already covered by `determineHostJsonFile` call.

**Review by sdubov — COMMENTED** [review]
This change looks good. However, we just silently stop execution from a user perspective. I assume we need to think about notification for those cases with the error reason. What do you think?

**Comment by maartenba** [comment]
Function names are usually empty, unless you use the gutter marks or edit the run configuration.
So if they are empty, I think we only need to show a warning in logs in case we need it, but other than that not touching `host.json` and running `func host` seems best (it's the default if you create a new project)

**Comment by maartenba** [comment]
Which leads me to think... Should we even log a warning... 🤔 Might make more sense to have it as an informational message, or even debug?

**Comment by sdubov** [comment]
`Info` should match the case. One other question that is related to this change - I think we need to add validation for attribute function name value. This is for the case when running from gutter mark. We might add a quick fix for this case. Maybe do it as a separate PR.

**Review by sdubov — APPROVED** [review]

**Pull request merged**